dapp/src/hooks/useDepositBTCTransaction.ts

```diff
@@ -1,65 +1,51 @@
-import { useCallback, useEffect, useState } from "react"
 import { OnErrorCallback, OnSuccessCallback } from "#/types"
+import { useMutation } from "@tanstack/react-query"
 import { useWallet } from "./useWallet"
 import { useConnector } from "./orangeKit/useConnector"
 
-type SendBitcoinTransactionParams = Parameters<
-  (recipient: string, amount: bigint) => void
->
-
-type UseDepositBTCTransactionReturn = {
-  transactionHash?: string
-  sendBitcoinTransaction: (
-    ...params: SendBitcoinTransactionParams
-  ) => Promise<void>
-}
-
 export function useDepositBTCTransaction(
-  onSuccess?: OnSuccessCallback,
+  onSuccess?: OnSuccessCallback<string>,
   onError?: OnErrorCallback,
```

Comment:
We can pass `onSuccess` and. `onError` to `mutate` function.

Reply:
https://github.com/thesis/acre/pull/821/commits/26da7a55c84e4bbd6f27156a5208ac63b04c5429